[BugFix] Fix Llama4 Pipeline Parallelism Assert Error

River12 opened the pull request:
Summary: When enabling PP for Llama4 model, each rank has a few `PPMissingLayer()`. The line `assert isinstance(layer, Llama4DecoderLayer)` raises assert error when encountering `PPMissingLayer()`. This PR fixed this bug by using if condition to check the layer is Llama4DecoderLayer.

Test Plan:
In OSS environment:
```
from vllm import LLM
llm = LLM("meta-llama/Llama-4-Scout-17B-16E-Instruct", pipeline_parallel_size=4)
output = llm.generate("Hi, vLLM is a")
```

Differential Revision: D86837484

hmellor reviewed (approved):
Thanks for the fix!

hmellor commented:
Please fix DCO and we can merge

hmellor commented:
Follow the instructions in https://github.com/vllm-project/vllm/pull/28577/checks?check_run_id=55275366558

River12 commented:
> Please fix DCO and we can merge

Thanks. DCO is added.

Commit:
Fix Llama4 Pipeline Parallelism Assert Error (#28577)

Summary:
Signed-off-by: Dezhan Tu <dztu@meta.com>

When enabling PP for Llama4 model, each rank has a few `PPMissingLayer()`. The line `assert isinstance(layer, Llama4DecoderLayer)` raises assert error when encountering `PPMissingLayer()`. This PR fixed this bug by skipping `PPMissingLayer()` in the loop of model layers.

Test Plan:
In OSS environment:
```
from vllm import LLM
llm = LLM("meta-llama/Llama-4-Scout-17B-16E-Instruct", pipeline_parallel_size=4)
output = llm.generate("Hi, vLLM is a")
```

Reviewed By: helunwencser

Differential Revision: D86837484

heheda12345 reviewed (approved):
LGTM!
